src/pedal/config.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
```

Comment:
Unless you have very weird requirements, `serde` can derive the `Deserialize` and `Serialize` traits for your structs/enums.

Reply:
`Deserialize` is a macro here, which is given the AST of the struct this is on, so it can look at the fields and their types and write an implementation of `Deserialize` for it.

Reply:
👍🏻